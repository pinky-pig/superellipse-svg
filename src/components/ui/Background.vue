<script setup lang="ts">
import { ref } from 'vue'
import '~/utils/RandomBg'

// 随机色色板
const palettes = [
  '031926-468189-77aca2-9dbebb-f4e9cd'.split('-').map(a => `#${a}`),
  'f4faff-dee7e7-b7adcf-4f646f-535657'.split('-').map(a => `#${a}`),
  'acadbc-9b9ece-6665dd-473bf0-000500'.split('-').map(a => `#${a}`),
  '88498f-779fa1-e0cba8-ff6542-564154'.split('-').map(a => `#${a}`),
  '493b2a-593f62-7b6d8d-8499b1-a5c4d4'.split('-').map(a => `#${a}`),
  'c41e3d-7d1128-ff2c55-3c0919-e2294f'.split('-').map(a => `#${a}`),
  '16bac5-5fbff9-efe9f4-171d1c-5863f8'.split('-').map(a => `#${a}`),
  'd9e5d6-00a7e1-eddea4-f7a072-ff9b42'.split('-').map(a => `#${a}`),
]

// 初始化颜色
const initial = [
  { path: 'polygon(2% 86%,4% 78%,26% 37%,24% 65%,11% 20%,15% 58%,3% 20%,2% 41%,9% 14%,18% 7%)', color: '#628395' },
  { path: 'polygon(55% 11%,44% 59%,47% 57%,59% 71%,58% 72%)', color: '#96897b' },
  { path: 'polygon(91% 1%,67% 87%,77% 0%,71% 45%)', color: '#dbad6a' },
]

// 只要改变这个值，就会触发重新渲染
const isRerenderFlag = ref(0)
function printf(e: any) {
  // console.log(e.detail, '重新渲染后的数据')
}

onMounted(() => {
  document.addEventListener('keydown', (event) => {
    if (event.key === ' ')
      isRerenderFlag.value = (isRerenderFlag.value + 1) % 11
  })
})
</script>

<template>
  <random-bg
    :palettes="palettes"
    :initial="initial"
    :rerender="`${isRerenderFlag}`"
    @rendered="printf"
  />
</template>
