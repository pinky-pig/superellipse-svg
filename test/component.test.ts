import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Background from '../src/components/ui/Background.vue'

describe('Background.vue', () => {
  it('测试随机背景', () => {
    const isRerenderFlag = 0
    const wrapper = mount(Background, { props: { isRerenderFlag } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
