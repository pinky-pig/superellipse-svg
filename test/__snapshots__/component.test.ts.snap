// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Background.vue > 测试随机背景 1`] = `"<random-bg palettes=\\"[[&quot;#031926&quot;,&quot;#468189&quot;,&quot;#77aca2&quot;,&quot;#9dbebb&quot;,&quot;#f4e9cd&quot;],[&quot;#f4faff&quot;,&quot;#dee7e7&quot;,&quot;#b7adcf&quot;,&quot;#4f646f&quot;,&quot;#535657&quot;],[&quot;#acadbc&quot;,&quot;#9b9ece&quot;,&quot;#6665dd&quot;,&quot;#473bf0&quot;,&quot;#000500&quot;],[&quot;#88498f&quot;,&quot;#779fa1&quot;,&quot;#e0cba8&quot;,&quot;#ff6542&quot;,&quot;#564154&quot;],[&quot;#493b2a&quot;,&quot;#593f62&quot;,&quot;#7b6d8d&quot;,&quot;#8499b1&quot;,&quot;#a5c4d4&quot;],[&quot;#c41e3d&quot;,&quot;#7d1128&quot;,&quot;#ff2c55&quot;,&quot;#3c0919&quot;,&quot;#e2294f&quot;],[&quot;#16bac5&quot;,&quot;#5fbff9&quot;,&quot;#efe9f4&quot;,&quot;#171d1c&quot;,&quot;#5863f8&quot;],[&quot;#d9e5d6&quot;,&quot;#00a7e1&quot;,&quot;#eddea4&quot;,&quot;#f7a072&quot;,&quot;#ff9b42&quot;]]\\" initial=\\"[{&quot;path&quot;:&quot;polygon(2% 86%,4% 78%,26% 37%,24% 65%,11% 20%,15% 58%,3% 20%,2% 41%,9% 14%,18% 7%)&quot;,&quot;color&quot;:&quot;#628395&quot;},{&quot;path&quot;:&quot;polygon(55% 11%,44% 59%,47% 57%,59% 71%,58% 72%)&quot;,&quot;color&quot;:&quot;#96897b&quot;},{&quot;path&quot;:&quot;polygon(91% 1%,67% 87%,77% 0%,71% 45%)&quot;,&quot;color&quot;:&quot;#dbad6a&quot;}]\\" rerender=\\"0\\" isrerenderflag=\\"0\\"></random-bg>"`;

exports[`TheCounter.vue > should render 1`] = `"<div>10 <button class=\\"inc\\"> + </button><button class=\\"dec\\"> - </button></div>"`;
